src/C++/SocketMonitor.h

```diff
@@ -61,10 +74,20 @@ class SocketMonitor
   void signal(socket_handle socket );
   void unsignal(socket_handle socket );
   void block( Strategy& strategy, bool poll = 0, double timeout = 0.0 );
+  void setSocketOperator(const std::shared_ptr<SocketOperator>& op ) {
+    socketOperator = op;
+  };
 
   size_t numSockets() 
   { return m_readSockets.size() - 1; }
 
+  void dropAllReadSockets(){
+    std::set<int> sockets = m_readSockets;
+    for(int sock : sockets) {
+      drop(sock);
+    }
+  }
+
```

Comment:
Helper method to drop all read sockets.  Added for 100% unit test coverage requirement as it is not possible to drop all sockets currently as there was an interrupt socket set via the constructor